crenel_caves.logic

```diff
@@ -1,17 +1,118 @@
-area CrenelCaves:
-  room BlocksChest
-  room BlocksPuzzle
-  room BlocksBomb
-  room FloodBlock
-  room FountHP
-  room FountRupee
-  room GraybladeEntrance
-  room Guru
-  room HelmaHall
-  room MushroomGap
-  room RockPush
-  room ScrubBomb
-  room ScrubGrip
-  room ScrubHint
-  room SwitchBridge
-  room PotCircle
\ No newline at end of file
+area CrenelCaves:
+  room BlocksChest:
+    node BlocksChest:
+      door Staircase
+      door Doorway
+      enemy RedChus
+      chest BlocksChest
+  room BlocksPuzzle:
+    node BlocksPuzzle:
+      door Ladder
+      door Doorway
+      enemy Helmasaurs
+    node MushroomGapSouth:
+      door Doorway
+      enemy Helmasaur
+    node MushroomGapNorth:
+      door Staircase
+      enemy Helmasaur
+    MushroomGapNorth <-> MushroomGapSouth: (item GustJar | item Cape | item LightArrows)
+  room BlocksBomb:
+    node PreBombBlocks:
+      door PreBombBlocks
+      enemy SilverChu
+      enemy RedChu
+    node PostBombBlocks:
+      door PostBombBlocks
+      enemy SilverChu
+    PreBombBlocks -> PostBombBlocks: item BombBag
+  room FloodBlock: #This is the room w/ 2 chests and a HP on the very left side of crenel right?
+    node FloodBlockEntrance:
+      door FloodBlock
+    node FloodBlockChests:
+      chest LeftFloodBlockChest
+      chest RightFloodBlockChest
+    FloodBlockEntrance -> FloodBlockChests: (item BombBag | item Cape)
+    node FloodBlockHP:
+      chest HearPiece
+    FloodBlockEntrance -> FloodBlockHP: (item BombBag | item Cape | item Flippers)
+  room FountHP:
+    node FountHP:
+      door FountHP
+      chest HeartPiece
+  room FountRupee:
+    node FountRupee:
+      door FountRupee
+    node FountRupeeItems:
+      chest LeftRupee
+      chest CenterRupee
+      chest RightRupee
+    FountRupee -> FountRupeeItems: setting Rupeemania
+  room GraybladeEntrance:
+    node GraybladeEntrance:
+      door Entrance
+    node Switches:
+	  flag Switches
+    #Completely guessing here on the actual item names/syntax
+	GraybladeEntrance -> Switches: (item SpinAttack & (item RedSword | item BlueSword | item FourSword))
+    node SwitchDoor:
+      door SwitchDoor
+    Switches <-> SwitchDoor

```

Comment:
This door might need to be split - it depends on how leaving greyblade without the flag for the door being set behaves. Either way, you don't set the flag for the door to be open by coming from the other side, so the connection here is wrong.